internal/ghinstance/host.go

```diff
@@ -6,37 +6,59 @@ import (
 	"strings"
 )
 
+// DefaultHostname is the domain name of the default GitHub instance.
 const defaultHostname = "github.com"
 
-// localhost is the domain name of a local GitHub instance
+// Localhost is the domain name of a local GitHub instance.
 const localhost = "github.localhost"
 
-// Default returns the host name of the default GitHub instance
+// TenancyHost is the domain name of a tenancy GitHub instance.
+const tenancyHost = "ghe.com"
+
+// Default returns the host name of the default GitHub instance.
 func Default() string {
 	return defaultHostname
 }
 
-// IsEnterprise reports whether a non-normalized host name looks like a GHE instance
+// IsEnterprise reports whether a non-normalized host name looks like a GHE instance.
 func IsEnterprise(h string) bool {
 	normalizedHostName := NormalizeHostname(h)
 	return normalizedHostName != defaultHostname && normalizedHostName != localhost
 }
 
+// IsTenancy reports whether a non-normalized host name looks like a tenancy instance.
+func IsTenancy(h string) bool {
+	normalizedHostName := NormalizeHostname(h)
+	return strings.HasSuffix(normalizedHostName, "."+tenancyHost)
+}
+
+// TenantName extracts the tenant name from tenancy host name.
+// Returns an empty string if host name is not a tenancy instance.
+func TenantName(h string) string {
+	normalizedHostName := NormalizeHostname(h)
+	if before, found := cutSuffix(normalizedHostName, "."+tenancyHost); found {
+		return before
+	}
+	return ""
+}
+
```

Comment:
One other way I could see this being written to avoid a footgun around "returns an empty string is host name is not a tenancy instance" would be to mimic the `cutSuffix` signature with something like:

```
func TenantName(h string) (bool, string)
```

In this way, you'd have to pretty explicitly avoid checking the `bool` to use the empty string incorrectly.

I don't feel strongly about this but wanted to throw it out there in case it resonated with you.